File: ios/Classes/BestktpocrflutterPlugin.swift
import Flutter
import UIKit
import OCRforEKTP
public class BestktpocrflutterPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "bestktpocrflutter", binaryMessenger: registrar.messenger())
    let instance = BestktpocrflutterPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    case "scanKTP":
        if let args = call.arguments as? Dictionary<String, Any> {
    
            guard let byteData = args["ktp"] as? FlutterStandardTypedData else {
                result("ktp must in Uint8List format")
                return
            }
            if let image = UIImage(data: Data([UInt8](byteData.data))) {
                let ocrLibrary = OCRforEKTPLibrary()
                let ocrResult = ocrLibrary.scanEKTP(image)
                let ktpData: [String: String] = [
                    "nik":  ocrResult.result.nik,
                    "nama": ocrResult.result.nama,
                    "tempatLahir": ocrResult.result.tempatLahir,
                    "tanggalLahir": ocrResult.result.tanggalLahir,
                    "jenisKelamin": ocrResult.result.jenisKelamin,
                    "alamat": ocrResult.result.alamat,
                    "rtrw": ocrResult.result.rtrw,
                    "kelurahan": ocrResult.result.kelurahan,
                    "kecamatan": ocrResult.result.kecamatan,
                    "agama": ocrResult.result.agama,
                    "statusPerkawinan": ocrResult.result.statusPerkawinan,
                    "pekerjaan": ocrResult.result.pekerjaan,
                    "kewarganegaraan": ocrResult.result.kewarganegaraan,
                    "provinsi": ocrResult.result.provinsi,
                    "kota": ocrResult.result.kota
                ]
                let ktpJsonData = try! JSONSerialization.data(withJSONObject: ktpData, options: [])
                let ktpJsonString = String(data: ktpJsonData, encoding: .utf8)!
                result(ktpJsonString)
            }else{
                result("ktp must in data format")
            }
        }else{
            result("no ktp found")
        }
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}

File: android/src/main/kotlin/id/co/tirtavium/bestktpocrflutter/BestktpocrflutterPlugin.kt
package id.co.tirtavium.bestktpocrflutter

import OCRforEKTPLibrary
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.AsyncTask
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import org.json.JSONObject
import java.util.concurrent.Executors


/** BestktpocrflutterPlugin */
class BestktpocrflutterPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "bestktpocrflutter")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    if (call.method == "getPlatformVersion") {
      result.success("Android ${android.os.Build.VERSION.RELEASE}")
    } else if (call.method == "scanKTP"){
      Executors.newSingleThreadExecutor().execute(Runnable {
        // todo: do your background tasks
      val bytes =  call.argument<ByteArray>("ktp");
      val image: Bitmap? = bytes?.let { BitmapFactory.decodeByteArray( bytes, 0, it.size) };
      //val image: Bitmap? = call.argument("ktp")
      val scanner = OCRforEKTPLibrary()
      val ktp = image?.let {
        val ktpData = scanner.scanEKTP(image)
        val ktpJson = JSONObject("""
          {
          "nik":"${ktpData.nik}", 
          "nama":"${ktpData.nama}",
          "tempatLahir":"${ktpData.tempatLahir}",
          "tanggalLahir":"${ktpData.tglLahir}",
          "jenisKelamin":"${ktpData.jenisKelamin}", 
          "alamat":"${ktpData.alamat}",
          "rtrw":"${ktpData.rtrw}",
          "kelurahan":"${ktpData.kelurahan}",
          "kecamatan":"${ktpData.kecamatan}",
          "agama":"${ktpData.agama}",
          "statusPerkawinan":"${ktpData.statusPerkawinan}",
          "pekerjaan":"${ktpData.pekerjaan}",
          "kewarganegaraan":"${ktpData.kewarganegaraan}",
          "provinsi":"${ktpData.provinsi}",
          "kota":"${ktpData.kabKot}"
          }
          """.trimIndent())
          result.success(ktpJson.toString())
      }

      })
    }
    else {
      result.notImplemented()
    }

  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
